lib/zip/ioextras/abstract_input_stream.rb

```diff
@@ -11,26 +11,27 @@ def initialize
         super
         @lineno        = 0
         @pos           = 0
-        @output_buffer = ''
+        @output_buffer = String.new
       end
 
       attr_accessor :lineno
       attr_reader :pos
 
       def read(number_of_bytes = nil, buf = '')
+        buffer = buf.dup
```

Comment:
Why don't you make `buf` a mutable `String.new('')` instead?

Reply:
reverted.